addon/components/new-osf-navbar/component.js

```diff
@@ -48,6 +48,10 @@ export default Ember.Component.extend(hostAppName, AnalyticsMixin, {
         const service = this.get('currentService');
         return this.get('serviceLinks')[serviceMapping[service]];
     }),
+    serviceSubmitLink: Ember.computed('serviceLinks', 'addPreprintUrl', function() {
```

Comment:
Putting this logic here makes it harder to figure out where the URLs actually come from. I think it'd be clearer to remove this computed property and pass `addPreprintUrl` directly to `build-secondary-nav-links`. At the same place it's setting `PREPRINTS.href`, it should choose between `addPreprintUrl` and `serviceLinks.preprintsSubmit`.